.github/workflows/build.yml

```diff
@@ -0,0 +1,80 @@
+---
+name: Build
+# Runs on any commit to master, or manually via the GitHub Actions UI
+on:
+  push:
+    branches:
+      - master
+  workflow_dispatch:  
+    
+jobs:
+  push_server:
+    name: Push Docker image for the API server to GitHub Packages
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Cache Docker build artifacts
+        uses: actions/cache@v2
+        with:
+          path: /tmp/buildx-cache
+          # Cache with a key based on the current Git commit
+          key: ${{ runner.os }}-server-buildx-${{ github.sha }}
+          # And restore from any available cache
+          restore-keys: |
+            ${{ runner.os }}-server-buildx-
+      - name: Build and push to GitHub Packages
+        uses: docker/build-push-action@v1
+        with:
+          context: thavalon-server
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          registry: docker.pkg.github.com
+          repository: theadd336/THavalonWeb/thavalon-server
```

Comment:
:crossed_fingers: this should work. I'm not entirely sure how the GitHub Packages setup works